File: app/src/main/java/nl/rsdt/japp/jotial/maps/searching/SearchEntry.java
package nl.rsdt.japp.jotial.maps.searching;

/**
 * @author Dingenis Sieger Sinke
 * @version 1.0
 * @since 29-7-2016
 * Description...
 */
public class SearchEntry {

    private String id;

    public String getId() {
        return id;
    }

    private int infoId;

    public int getInfoId() {
        return infoId;
    }

    private String value;

    public String getValue() {
        return value;
    }

    public static class Builder
    {
        SearchEntry entry = new SearchEntry();

        public Builder setId(String string)
        {
            entry.id = string;
            return this;
        }

        public Builder setInfoId(int id)
        {
            entry.infoId = id;
            return this;
        }

        public Builder setValue(String value) {
            entry.value = value;
            return this;
        }

        public SearchEntry create()
        {
            return entry;
        }

    }



}

File: app/src/main/java/nl/rsdt/japp/jotial/maps/searching/SearchEntry.kt
package nl.rsdt.japp.jotial.maps.searching

/**
 * @author Dingenis Sieger Sinke
 * @version 1.0
 * @since 29-7-2016
 * Description...
 */
class SearchEntry {

    var id: String? = null
        private set

    var infoId: Int = 0
        private set

    var value: String? = null
        private set

    class Builder {
        internal var entry = SearchEntry()

        fun setId(string: String): Builder {
            entry.id = string
            return this
        }

        fun setInfoId(id: Int): Builder {
            entry.infoId = id
            return this
        }

        fun setValue(value: String): Builder {
            entry.value = value
            return this
        }

        fun create(): SearchEntry {
            return entry
        }

    }


}
